src/main/java/org/wildfly/security/audit/PeriodicRotatingFileAuditEndpoint.java

```diff
@@ -0,0 +1,226 @@
+                        .withDayOfMonth(1)
+                        .plus(1,ChronoUnit.MONTHS);
+                break;
+            case WEEK:
+                zonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.DAYS)
+                        .with(TemporalAdjusters.next(WeekFields.of(Locale.getDefault()).getFirstDayOfWeek()));
+                break;
+            case DAY:
+                zonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.DAYS)
+                        .plus(1, ChronoUnit.DAYS);
+                break;
+            case HALF_DAY:
+                ZonedDateTime halfDay = ZonedDateTime.from(zonedDateTime).truncatedTo(ChronoUnit.DAYS)
+                        .plus(1, ChronoUnit.HALF_DAYS);
+                if ( zonedDateTime.isBefore(halfDay) ) {
+                    zonedDateTime = halfDay;
+                }else{
+                    zonedDateTime = halfDay.plus(1, ChronoUnit.HALF_DAYS);
+                }
+                break;
+            case HOUR:
+                zonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.HOURS)
+                        .plus(1, ChronoUnit.HOURS);
+                break;
+            case MINUTE:
+                zonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.MINUTES)
+                        .plus(1, ChronoUnit.MINUTES);
+        }
+        nextRollover = zonedDateTime.toInstant().toEpochMilli();
+    }
+
+    /**
+     * Possible period values. Keep in strictly ascending order of magnitude.
+     */
+    protected enum Period {
+        MINUTE,
+        HOUR,
+        HALF_DAY,
+        DAY,
+        WEEK,
+        MONTH,
+        YEAR,
+        NEVER,
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder extends FileAuditEndpoint.Builder {
+        DateTimeFormatter format;
+        Period period = Period.NEVER;
+        ZoneId timeZone = ZoneId.systemDefault();
+
+        Builder() {
+            super();
+        }
+
+        /**
+         * Set the configured time zone for this handler.
+         *
+         * @param timeZone the configured time zone
+         * @return this builder.
+         */
+        public Builder setTimeZone(ZoneId timeZone) {
+            this.timeZone = checkNotNullParam("timeZone", timeZone);
+
+            return this;
+        }
+
+        /**
+         * Set the suffix string.  The string is in a format which can be understood by {@link java.time.format.DateTimeFormatter}.
+         * The period of the rotation is automatically calculated based on the suffix.
+         * The following characters \/:*?"<>| are always replaced by '_'
+         *
+         * @param suffix the suffix
+         * @throws IllegalArgumentException if the suffix is not valid
+         */
+        public Builder setSuffix(String suffix) throws IllegalArgumentException {
+            format = DateTimeFormatter.ofPattern(suffix.replaceAll("[\\/:*?\"<>|]", "_")).withZone(timeZone);
```

Comment:
+1 This does seem a little over restrictive.

Also at some point this kind of check may be better handled in the management model, i.e. in the model we can either perform MODEL stage validation or correct values.